zipkin-server/src/main/java/zipkin/server/ZipkinServer.java

```diff
@@ -13,17 +13,70 @@
  */
 package zipkin.server;
 
+import com.nike.riposte.server.Server;
+import com.nike.riposte.server.config.ServerConfig;
+import com.nike.riposte.server.http.Endpoint;
+import com.nike.riposte.server.http.filter.RequestAndResponseFilter;
+import com.nike.riposte.server.http.filter.impl.AllowAllTheThingsCORSFilter;
+
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 import zipkin.server.brave.BootstrapTrace;
 
+import static java.util.Collections.singletonList;
+
 @SpringBootApplication
 @EnableZipkinServer
 public class ZipkinServer {
 
-  public static void main(String[] args) {
-    new SpringApplicationBuilder(ZipkinServer.class)
+  public static void main(String[] args) throws CertificateException, InterruptedException, IOException {
+    ConfigurableApplicationContext cac = new SpringApplicationBuilder(ZipkinServer.class)
```

Comment:
The regular Spring Boot app is still running normally, this PoC just adds a Riposte server listening on port 8042 that uses the same Spring `ApplicationContext` as the Spring Boot server. I verified that I could send spans to `/api/v1/spans` on either the Spring Boot app port 9411 or the Riposte port 8042 and they would both show up in the UI.

A full conversion that fully replaced Spring Boot might do something like the commented out lines below for starting a Spring `ApplicationContext`, but the existing Spring dependency injection wiring would stay mostly (entirely?) unchanged.